File: book_app_api/app/entities/book.py
from sqlalchemy import ForeignKey, String, Enum, LargeBinary
from app.core.database_config import Base
from sqlalchemy.orm import relationship, mapped_column, Mapped
from typing import List

class Book(Base):
    __tablename__ = "book"
    id: Mapped[int] = mapped_column(primary_key = True, autoincrement=True)
    book_url: Mapped[str] = mapped_column(nullable=False)
   
    genre: Mapped[str] = mapped_column(nullable=True)
    description: Mapped[str] = mapped_column( nullable=True)
    cover_data: Mapped[bytes] = mapped_column(LargeBinary, nullable=True)
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    author: Mapped[str] = mapped_column(String(255), nullable=False)
    date: Mapped[str] = mapped_column(nullable=True)
    language: Mapped[str] = mapped_column(nullable=True)

    progress: Mapped[List["Progress"]] = relationship("Progress",back_populates="book",lazy="select")
    

File: book_app_fe/lib/features/books/domain/entities/book.dart
import 'dart:convert';
import 'dart:typed_data';

class Book {
  final int id;
  final String title;
  final String author;
  final String? description;
  final String? coverBase64;
  final DateTime? date;
  final String? language;
  final String? genre;

  Book({
    required this.id,
    required this.title,
    required this.author,
    this.description,
    this.coverBase64,
    this.date,
    this.language,
    this.genre,
  });

  Uint8List? get decodedCover {
    if (coverBase64 == null) return null;
    return base64Decode(coverBase64!);
  }
}
